File: JavaScript_Basic_04_2023_SolutionsOfProblems/06.LAB_ PB-JS-Nested-Loops/P04.SumOfTwoNumbers.js
function sumOfTwoNumbers(input) {
    let startNumber = Number(input[0]);
    let endNumber = Number(input[1]);
    let magicNumber = Number(input[2]);
    let counter = 0;
    let isFindMagicSum = false;

    for (let i = startNumber; i <= endNumber; i++) {
        for (let j = startNumber; j <= endNumber; j++) {
            counter++;
            if (i + j === magicNumber) {
                console.log(`Combination N:${counter} (${i} + ${j} = ${magicNumber})`);
                isFindMagicSum = true;
                break;
            }
        }
        if (isFindMagicSum) {
            break;
        }
    }
    if (!isFindMagicSum) {
        console.log(`${counter} combinations - neither equals ${magicNumber}`);
    }
}
sumOfTwoNumbers(["88", 
"888", 
"2000"])

File: Java_Basic_Jan_2023_SolutionsOfProblems/P06LAB_Nested_Loop/P04_SumOfTwoNumbers.java
package P06LAB_Nested_Loop;

import java.util.Scanner;

public class P04_SumOfTwoNumbers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int minNumber = Integer.parseInt(scanner.nextLine());
        int maxNumber = Integer.parseInt(scanner.nextLine());
        int magicNumber = Integer.parseInt(scanner.nextLine());
        int combinations = 0;
        boolean founded = false;
        for (int n1 = minNumber; n1 <= maxNumber; n1++) {
            for (int n2 = minNumber; n2 <= maxNumber; n2++) {
                combinations++;
                if (n1 + n2 == magicNumber) {
                    System.out.printf("Combination N:%d (%d + %d = %d)", combinations, n1, n2, magicNumber);
                    founded = true;
                    break;
                }
            }
            if (founded) {
                break;
            }
        }
        if (!founded) {
            System.out.printf("%d combinations - neither equals %d", combinations, magicNumber);
        }
    }
}
